shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm

```diff
@@ -471,79 +471,83 @@
 
   DisposeViews();
 
-  SkCanvas* background_canvas = frame->SkiaCanvas();
+  auto did_submit = true;
+  auto num_platform_views = composition_order_.size();
 
-  // Resolve all pending GPU operations before allocating a new surface.
-  background_canvas->flush();
-  // Clipping the background canvas before drawing the picture recorders requires to
-  // save and restore the clip context.
-  SkAutoCanvasRestore save(background_canvas, /*doSave=*/true);
   // Maps a platform view id to a vector of `FlutterPlatformViewLayer`.
   LayersMap platform_view_layers;
 
-  auto did_submit = true;
-  auto num_platform_views = composition_order_.size();
-
-  for (size_t i = 0; i < num_platform_views; i++) {
-    int64_t platform_view_id = composition_order_[i];
-    sk_sp<RTree> rtree = platform_view_rtrees_[platform_view_id];
-    sk_sp<SkPicture> picture = picture_recorders_[platform_view_id]->finishRecordingAsPicture();
-
-    // Check if the current picture contains overlays that intersect with the
-    // current platform view or any of the previous platform views.
-    for (size_t j = i + 1; j > 0; j--) {
-      int64_t current_platform_view_id = composition_order_[j - 1];
-      SkRect platform_view_rect = GetPlatformViewRect(current_platform_view_id);
-      std::list<SkRect> intersection_rects =
-          rtree->searchNonOverlappingDrawnRects(platform_view_rect);
-      auto allocation_size = intersection_rects.size();
-
-      // For testing purposes, the overlay id is used to find the overlay view.
-      // This is the index of the layer for the current platform view.
-      auto overlay_id = platform_view_layers[current_platform_view_id].size();
-
-      // If the max number of allocations per platform view is exceeded,
-      // then join all the rects into a single one.
-      //
-      // TODO(egarciad): Consider making this configurable.
-      // https://github.com/flutter/flutter/issues/52510
-      if (allocation_size > kMaxLayerAllocations) {
-        SkRect joined_rect;
-        for (const SkRect& rect : intersection_rects) {
-          joined_rect.join(rect);
+  if (num_platform_views > 0) {
```

Comment:
cool. Android has this check too.  Any chance we add a test for this?

Reply:
No idea about how to add a test. :)

Reply:
@blasten Can you answer this query so progress is unblocked here?